### DISCUSSION TITLE
minio memory keeps increasing

### DISCUSSION THREAD

**Discussion by sathishkumar-p** [minio/minio#17871] [Q&A]
## minio memory keeps increasing
I deployed minio with 4 drivers and 4 servers clusters with help of minio operator. Allocated 4GB of Memory for each server. I can see there is gradually increasing the RAM usage. Once its reaches the limit the POD crash and post restart the memory usage will be 500MB from GB.

## Expected Behavior
Garbage collector not cleaning the memory of the minio pod. 

## Current Behavior
Minio POD memory is gradually increasing even when its unused.

## Steps to Reproduce (for bugs)
1. Deployed minio 4 drivers and 4 servers
2. Wait for a few days
3. you can see the ram usage high

## Your Environment
* Version used (`minio --version`): RELEASE.2023-06-23T20-26-00Z 
* Server setup and configuration: Kubernetes Minio operator 4 server and 4 drivers per node
* Operating System and version (`uname -a`): Kubernetes cluster

**Comment by harshavardhana** [comment]
@sathishkumar-p please capture `mc support profile` output

**Reply by sathishkumar-p** [reply]
Hi @harshavardhana ,
Sry for the late reply.

[profile](https://drive.google.com/file/d/1bDBLKx7zbrmpH3RMIz2-6AEL14QDxEgu/view?usp=drive_link)

**Reply by sathishkumar-p** [reply]
Gave access to the profile. Please try now

**Comment by harshavardhana** [comment]
Also please use the latest release RELEASE.2023-08-16T20-17-30Z and then report the issue.

**Reply by sathishkumar-p** [reply]
Sure i will upgrade it. But will it solve the issue ?

**Reply by harshavardhana** [reply]
@sathishkumar-p I won't be able to confirm that we have no idea about your cluster.

**Comment by harshavardhana** [comment]
MinIO's active memory is only 160MiB - from what I can tell there is something else that is perhaps eating up such as not using aggressive GC. 

Please set `export GC=10` to ensure GC kicks in faster for low memory requirements that you have.

```
for i in $(ls *-mem.pprof); do go tool pprof -text profile-minio-pool-0-1.minio-hl.minio.svc.cluster.local\:9000-mem.pprof | head -5; done | grep 'nodes'
Showing nodes accounting for 165693.44kB, 92.33% of 179465.95kB total
Dropped 1007 nodes (cum <= 897.33kB)
Showing nodes accounting for 165693.44kB, 92.33% of 179465.95kB total
Dropped 1007 nodes (cum <= 897.33kB)
Showing nodes accounting for 165693.44kB, 92.33% of 179465.95kB total
Dropped 1007 nodes (cum <= 897.33kB)
Showing nodes accounting for 165693.44kB, 92.33% of 179465.95kB total
Dropped 1007 nodes (cum <= 897.33kB)
Showing nodes accounting for 165693.44kB, 92.33% of 179465.95kB total
Dropped 1007 nodes (cum <= 897.33kB)
Showing nodes accounting for 165693.44kB, 92.33% of 179465.95kB total
Dropped 1007 nodes (cum <= 897.33kB)
Showing nodes accounting for 165693.44kB, 92.33% of 179465.95kB total
Dropped 1007 nodes (cum <= 897.33kB)
Showing nodes accounting for 165693.44kB, 92.33% of 179465.95kB total
Dropped 1007 nodes (cum <= 897.33kB)
```

**Reply by sathishkumar-p** [reply]
GC=10 is it a environment variables for minio ?

**Reply by R-Studio** [reply]
@sathishkumar-p have you found a solution?

**Reply by jerryguo8** [reply]
Any solutions for this? I have same issues for using `minio operator`. we have 4 pod deployed in 4 node of k8s cluster.  each of the pod memory usage can jump to 32GB for uploading 2 million of 25kb items.. the memory usage only drop when put operation is completed.
